src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
-        boolean needNotify = ! canWrite();
-        this.addToWriteSize(bb.remaining());
-        writeBuffers.add(bb.slice());
-        if(needNotify) {
-          this.flagWriteable();
-        }
-    }
-  }
-
+  public void writeForce(ByteBuffer bb);
+  
   /**
-   * This is called by the SocketExecuter once the clients socket is able to be written to.
-   * It is up to the clients implementer to figure out how to deal with combining or not combining ByteBuffers 
-   * for write.  In general its much better to just send the byteBuffers as they where received but the write functions. 
-   * 
-   * @return a ByteBuffer that can be used to Read new data off the socket for this client.
-   */
-  protected ByteBuffer getWriteBuffer() {
-    synchronized(writeBuffers) {
-      //This is to keep from doing a ton of little writes if we can.  We will try to 
-      //do at least 8k at a time, and up to 65k if we are already having to combine buffers
-      if(currentWriteBuffer == null || currentWriteBuffer.remaining() == 0) {
-        if(writeBuffers.nextPopSize() < 65536/8 && writeBuffers.remaining() > writeBuffers.nextPopSize()) {
-          if(writeBuffers.remaining() < 65536) {
-            currentWriteBuffer = writeBuffers.pull(writeBuffers.remaining());
-          } else {
-            currentWriteBuffer = writeBuffers.pull(65536);
-          }
-        } else {
-          currentWriteBuffer = writeBuffers.pop();
-        }
-      }
-      return currentWriteBuffer;
-    }
-  }
+   * <p>This provides the next available Write buffer.  This is typically only called by the {@link SocketExecuterInterface}.
+   * This is not threadsafe as the same {@link ByteBuffer} will be provided to any thread that calls this, until 
+   * something consumes all byte available in it at which point another {@link ByteBuffer} will be provided.</p>
+   * 
+   * <p>Assuming something actually removes data from the returned {@link ByteBuffer} an additional call to 
+   * {@link #reduceWrite(int)} must be made</p>
+   * 
+   * 
+   * @return a {@link ByteBuffer} that can be used to Read new data off the socket for this client.
+   */
+  public ByteBuffer getWriteBuffer();
   
   /**
-   * This is called after a write is written to the clients socket.  This tells the client how much of that ByteBuffer was written.
-   * This should also call removeFromWriteSize(size). to reduce the buffer and update stats.
+   * <p>This is called after a write is written to the clients socket.  This tells the client how much of that 
+   * {@link ByteBuffer} was written and then reduces the writeBuffersSize accordingly.</p>
    * 
    * @param size the size in bytes of data written on the socket.
    */
-  protected void reduceWrite(int size) {
-    synchronized(writeBuffers) {
-      removeFromWriteSize(size);
-      if(! currentWriteBuffer.hasRemaining()) {
-        currentWriteBuffer = null;
-      }
-      writeBuffers.notifyAll();
-    }
-  }
+  public void reduceWrite(int size);
```

Comment:
if they implemented there own Client they would for sure need this.  This gets very tricky because technically I could look at the write buffer I have them when they called getWriteBuffer and use its .remaining + all other write BB's, but I have found it easier and faster for the writer to just tell me how much it wrote.

Let me know if you think I should change it because it is totally doable.